site/lib/main.dart

```diff
@@ -0,0 +1,161 @@
+        MemoryLoader(pages: allMemoryPages),
+      ],
+      configResolver: PageConfig.all(
+        dataLoaders: [
+          FilesystemDataLoader(path.join(siteSrcPath, 'data')),
+          DataProcessor(),
+        ],
+        templateEngine: LiquidTemplateEngine(
+          includesPath: path.join(
+            siteSrcPath,
+            // TODO(parlough): Figure out why liquid is going back a directory,
+            //  requiring this to be duplicated.
+            '_includes',
+            '_includes',
+          ),
+        ),
+        parsers: const [
+          DashMarkdownParser(),
+          HtmlParser(),
+        ],
+        rawOutputPattern: RegExp(r'.*\.txt$'),
+        extensions: allNodeProcessingExtensions,
+        components: [
+          const DashTabs(),
+          CustomComponent(
+            pattern: RegExp('ArchiveTable', caseSensitive: false),
+            builder: (name, attributes, child) {
+              final channel = attributes['channel']!;
+              return ArchiveTable(channel: channel);
+            },
+          ),
+          CustomComponent(
+            pattern: RegExp('LintRuleIndex', caseSensitive: false),
+            builder: (name, attributes, child) {
+              return const LintRuleIndex();
+            },
+          ),
+          CustomComponent(
+            pattern: RegExp('DiagnosticIndex', caseSensitive: false),
+            builder: (name, attributes, child) {
+              return const DiagnosticIndex();
+            },
+          ),
+          CustomComponent(
+            pattern: RegExp('Card', caseSensitive: false),
+            builder: (name, attributes, child) {
+              final link = attributes['link'];
+              final title = attributes['title']!;
+              return Card(
+                header: [
+                  header(classes: 'card-title', [text(title)]),
+                ],
+                content: [?child],
+                link: link,
+                filled: link != null,
+              );
+            },
+          ),
+          CustomComponent(
+            pattern: RegExp('YouTubeEmbed', caseSensitive: false),
+            builder: (name, attributes, child) {
+              final rawVideoId = attributes['id'] as String;
+              final videoTitle = attributes['title'] as String;
+              final playlistId = attributes['playlist'];
+
+              final String videoId;
+              final int startTime;
+              if (rawVideoId.contains('?')) {
+                videoId = rawVideoId.split('?')[0];
+
+                final idAndStartTime = videoId.split('start=');
+                startTime = int.parse(idAndStartTime[1]);
+              } else {
+                startTime = 0;
+                videoId = rawVideoId;
+              }
+
+              return raw('''
+<lite-youtube videoid="$videoId" videotitle="$videoTitle" videoStartAt="$startTime" ${playlistId != null ? 'playlistid="$playlistId"' : ''}>
```

Comment:
Is this for getting YT embeds up and running faster? (I'm just going off the package description.)